### PR TITLE
Feature/hexxle 78 UI score

### PR DESCRIPTION
**Pull request by DennisBannerman**

### CODE REVIEW

File: unity/Assets/Scripts/Interfaces/IGameScore.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assets.Scripts.Interfaces
+{
+    public interface IGameScore
+    {
+        void IncreaseScore(int amount);
+        int GetCurrentScore();
+        int GetNextScoreThreshold();
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Interfaces/IGameScore.cs:13]: I assume the Threshold is preparation for adding new tiles to the stack?

File: unity/Assets/Scripts/Logic/Score.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            return _nextThreshold;
+        }
+
+        public void IncreaseScore(int amount)
+        {
+            _score += amount;
+            _score = Math.Max(_lastThreshold, _score);
+            if (_score >= _nextThreshold)
+            {
+                // TODO: implement interval logic
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Logic/Score.cs:38]: Is this for future use or was this forgotten?

File: unity/Assets/Scripts/Unity/UnityMap.cs [hunk 3]
```diff
@@ -32,7 +33,8 @@ private void Awake()
 
         private void Start()
         {
-            unityStack = GameObject.Find("Game").GetComponent("UnityStack") as UnityStack;
+            unityStack = GameObject.FindGameObjectWithTag("Stack").GetComponent<UnityStack>();
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Unity/UnityMap.cs:36]: We should probably explain the tag mechanism in the next meeting, might be usefull for other tasks aswell

**Comment by RicardoMonteiroSimoes** [comment]
seems to work, new record? 😄 
image

File: unity/Assets/Scripts/Unity/UnityMap.cs [hunk 4]
```diff
@@ -142,12 +146,12 @@ private void OnTilePlaced(object sender, TileMapEventArgs<ITile> e)
...
             }
 
-            Instantiate(
+            var newTile = Instantiate(
                     template,
                     CoordinateToPoint(tile.Coordinate),
                     Quaternion.Euler(-90, 0, 0)
                 );
-
+            newTile.transform.parent = this.transform;
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Unity/UnityMap.cs:154]: what does this exactly do? This seems to be unity related but I cant quite find anything on it

File: unity/Assets/Scripts/Unity/UnityPoints.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        IGameScore score;
+        public TMP_Text pointsText;
+
+        private void Awake()
+        {
+            score = new Score();
+        }
+
+        // Start is called before the first frame update
+        void Start()
```
Comment (RicardoMonteiroSimoes) [inline unity/Assets/Scripts/Unity/UnityPoints.cs:21]: recommend removing unused code

**Review by RicardoMonteiroSimoes — CHANGES_REQUESTED** [review]
Okay, it seems as if the gitignore does include everything in the assets folder. Recommendation:

Adapt gitignore so the `manifest.json` inside the `Assets` folder is versioned, but everything else isnt.

File: unity/Assets/Scripts/Interfaces/IGameScore.cs (hunk 1 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Interfaces/IGameScore.cs:13]: Yes, this is in preparation for the score based intervals which will allow you to gain additional tiles for the stack.

File: unity/Assets/Scripts/Logic/Score.cs (hunk 2 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Logic/Score.cs:38]: This is for future use. This is the place to attach further logic relating to additional tiles.

File: unity/Assets/Scripts/Unity/UnityMap.cs (hunk 3 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityMap.cs:36]: Will do.

File: unity/Assets/Scripts/Unity/UnityMap.cs (hunk 4 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityMap.cs:154]: Remember when we had every single tile at the root of the object tree in unity? This makes new tiles children of the map - should any further bugs like the one in the last meeting rise up, I'm sure, we're going to be happy about some structure in the object tree.

File: unity/Assets/Scripts/Unity/UnityPoints.cs (hunk 5 above)
Comment (DennisBannerman) [reply unity/Assets/Scripts/Unity/UnityPoints.cs:21]: Fair enough.

**Review by RicardoMonteiroSimoes — APPROVED** [review]

**Pull request merged**